src/libs/antares/inifile/inifile.cpp

```diff
@@ -253,51 +160,110 @@ uint IniFile::Section::size() const
     return count;
 }
 
-bool IniFile::open(const AnyString& filename, bool warnings)
+static bool startingSection(std::string line)
 {
-    // clear
-    clear();
-    pFilename = filename;
+    boost::trim(line);
+    return line.starts_with("[") && line.ends_with("]");
+}
 
-    // Load the file
-    IO::File::Stream file;
-    if (file.open(filename))
+static std::string getSectionName(std::string line)
+{
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("[]"));
+    return splitLine[1];
+}
+
+static bool isProperty(std::string line)
+{
+    return std::ranges::count(line.begin(), line.end(), '=') == 1;
+}
+
+static std::pair<std::string, std::string> getKeyValuePair(std::string line)
+{
+    boost::trim(line);
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("="));
+
+    return std::make_pair(splitLine[0], splitLine[1]);
+}
+
+static bool isComment(std::string line)
+{
+    boost::trim_left(line);
+    return line.starts_with("#") || line.starts_with(";");
+}
+
+bool isEmpty(std::string line)
+{
+    boost::trim(line);
+    return line.empty();
+}
+
+bool IniFile::readStream(std::istream& in_stream)
+{
+    std::string line;
+    IniFile::Section* currentSection(nullptr);
+    uint64_t read = 0;
+    while (std::getline(in_stream, line))
     {
-        enum
+        read += line.size();
+        if (startingSection(line))
         {
-            lineHardLimit = 2048
-        };
-
-        auto* line = new char[lineHardLimit];
+            currentSection = addSection(getSectionName(line));
+            continue;
+        }
 
-        IniFile::Section* lastSection = nullptr;
-        uint64_t read = 0;
+        if (isProperty(line))
+        {
+            // Note : if a property not in a section, it's simply skipped
```

Comment:
Why not get an error instead?
Was i the previous behaviour ?

Reply:
I first raised an error in the code when a property was not inside a section.
But I realized that the previous behavior was to skip such properties.
That's why it's emphasized in the code as a comment.